lib/index.js

```diff
@@ -110,7 +108,10 @@ module.exports = {
 
 						// handle "unused" mode
 						if (!noneUnusedMode) {
-							let scope = context.getScope();
+							let sourceCode = context.sourceCode ?? context.getSourceCode();
```

Comment:
seems like we're checking these things repeatedly (basically, determining ESLint 9+ versus 8 and below)... wondering if we could test once, and centralize...

for example:

```js
var getScope = (context,sourceCode) => {
   getScope = (
      sourceCode.getScope ?
         (context,sourceCode) => sourceCode.getScope() :
         (context,sourceCode) => context.getScope()
   );
   return getScope(context,sourceCode);
```

Here, we lazily do the test the first time `getScope(..)` is called, then overwrite the `getScope(..)` function so it doesn't need to do the checks any more throughout that run.

And we could do similar for the "sourceCode" and "ancestors" tests/evaluations.

Perhaps an approach like this reduces the code AND is *slightly* more performant?

Reply:
I have pushed the changes
Let me know if I misunderstood what you meant